File: 509.fibonacci-number.cs
// @leet start
public class Solution {
    private Dictionary<int, int> memory = new Dictionary<int, int>();
    public int Fib(int n) {
      if (n < 2) return n;
      if (memory.ContainsKey(n)) return memory[n];
      memory[n] = Fib(n-1) + Fib(n-2);
      return memory[n];
    }
}
// @leet end

File: 509.fibonacci-number.ts
// @leet start
function fib(n: number): number {
  if (n <= 1) return n
  let a = 0, b = 1
  for (let i = 2; i <= n; i++) {
    b = a + b
    a = b - a
  }
  return b
};
// @leet end
